src/metabase/pulse/render/datetime.clj

```diff
@@ -11,33 +11,107 @@
            java.time.Period
            java.time.temporal.Temporal))
 
+(defn temporal-string?
+  "Returns `true` if the string `s` is parseable as a datetime.
+
+  `(temporal-string? \"asdf\")` -> false
+  `(temporal-string? \"2020-02-02\")` -> true"
+  [s]
+  (boolean
+    (try
+      (u.date/parse s)
+      (catch Exception _e false))))
+
```

Comment:
Use this to make sure legend labels like "asdf" don't get put through the temporal string formatting. Open to a better way than to use a try catch.